src/mono/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.xml

```diff
@@ -309,7 +309,9 @@
 
 		<!-- domain.c: mono_defaults.runtimetype_class -->
 		<!-- under mono, this has no runtime visible fields -->
```

Comment:
Can we update this comment. It sort of doesn't make sense anymore.

Reply:
Are you just expecting a comment explaining why the constructor is needed?

Does someone from the mono team have a good explanation for why its actually needed here?

Reply:
Yes, I'd expect a comment on why the constructor is needed. We can probably leave the part about not having runtime visible fields, to help explain why no fields are being rooted.

Reply:
It would probably be good if someone from the mono or linker team provided the explanation for me to add here.

Otherwise its just going to end up as something like "RuntimeType is a fundamental type and rooting the constructor is required to avoid the linker thinking it can't be instanatiated"

Reply:
RuntimeType is a fundamental type and the constructor is called by the runtime to implement `Object.GetType` and `typeof(C)`

Reply:
```suggestion
		<!-- under mono, this has no runtime visible fields -->
		<!-- RuntimeType is a fundamental type and the constructor is called by the runtime to implement reflection primitives like Object.GetType() and typeof(C) -->
```

Reply:
The code calling it looks like https://github.com/dotnet/runtime/blob/master/src/mono/mono/metadata/reflection.c#L526 and is probably hit in the `if (typeof(TKey) == typeof(string))` check in the dictionary ctor